scripts/configure

```diff
@@ -26,98 +14,186 @@ check_dependencies () {
+# Configure for mainnet or testnet or regtest depending
+# upon the user-supplied value of $NETWORK
+BITCOIN_CHAIN="${NETWORK:-mainnet}"
+
+if [ "$BITCOIN_CHAIN" != "mainnet" ] && [ "$BITCOIN_CHAIN" != "testnet" ] && [ "$BITCOIN_CHAIN" != "regtest" ]; then
+  echo "Error: Umbrel can only be configured for mainnet (default), testnet or regtest"
+  exit 1
 fi
-# REGTEST set and TESTNET not
-if [ -z $TESTNET ] && [ ! -z $REGTEST ]; then
-    echo "Enabling regtest mode if REGTEST variable is set"
-    sed -i 's/\#\[regtest\]/\[regtest\]/g;' bitcoin/bitcoin.conf
-    sed -i 's/\#regtest=1/regtest=1/g' bitcoin/bitcoin.conf
-    sed -i 's/rpcport=8332/rpcport=18443/g; ' bitcoin/bitcoin.conf
-    sed -i 's/port=8333/port=18444/; ' bitcoin/bitcoin.conf
-    sed -i 's/mainnet/regtest/g; ' docker-compose.yml
-    echo "Setting regtest port"
-    sed -i 's/RPCPORT/18443/g; ' docker-compose.yml
-    # Update LND
-    echo "Changing LND to regtest mode"
-    sed -i 's/bitcoin.mainnet=1/bitcoin.regtest=1/g; ' lnd/lnd.conf
-    echo "Updating LND if regtest is set"
-    sed -i 's/bitcoin.node=neutrino/bitcoin.node=bitcoind/g; ' lnd/lnd.conf
+
+echo
+echo "Configuring Umbrel for $BITCOIN_CHAIN"
+echo 
+
+
+##########################################################
+############### Setup configuration files ###############
+##########################################################
+
+# Store paths to intermediary config files
+BITCOIN_CONF_FILE="./templates/bitcoin.conf"
+LND_CONF_FILE="./templates/lnd.conf"
+TOR_CONF_FILE="./templates/torrc"
+ENV_FILE="./templates/.env"
+
+# Remove intermediary files if they exist from any
+# previous unclean configuration run
+[[ -f "$BITCOIN_CONF_FILE" ]] && rm -f "$BITCOIN_CONF_FILE"
+[[ -f "$LND_CONF_FILE" ]] && rm -f "$LND_CONF_FILE"
+[[ -f "$TOR_CONF_FILE" ]] && rm -f "$TOR_CONF_FILE"
+[[ -f "$ENV_FILE" ]] && rm -f "$ENV_FILE"
+
+# Copy template configs to intermediary configs
+[[ -f "./templates/bitcoin-sample.conf" ]] && cp "./templates/bitcoin-sample.conf" "$BITCOIN_CONF_FILE"
+[[ -f "./templates/lnd-sample.conf" ]] && cp "./templates/lnd-sample.conf" "$LND_CONF_FILE"
+[[ -f "./templates/torrc-sample" ]] && cp "./templates/torrc-sample" "$TOR_CONF_FILE"
+[[ -f "./templates/.env-sample" ]] && cp "./templates/.env-sample" "$ENV_FILE"
+
+
+##########################################################
+############ Generate configuration variables ############
+##########################################################
+
+# Generate RPC credentials
+echo "Generating auth credentials"
+echo
+BITCOIN_RPC_USER="umbrelrpc"
+BITCOIN_RPC_DETAILS=$("./scripts/rpcauth.py" "$BITCOIN_RPC_USER")
+BITCOIN_RPC_AUTH=$(echo "$BITCOIN_RPC_DETAILS" | head -2 | tail -1)
+BITCOIN_RPC_PASS=$(echo "$BITCOIN_RPC_DETAILS" | tail -1)
+BITCOIN_RPC_PORT=8332
+BITCOIN_P2P_PORT=8333
+
+# Pull Tor image and generate Tor password
+echo "Generating Tor password"
+echo
+docker pull --quiet getumbrel/tor:v0.4.1.9
```

Comment:
should try out ```lncm/tor:0.4.3.6``` its completely built from source and supports multiple architectures

Reply:
Sounds good, we can do that in another PR after trying it out as it's non-related to the configuration-logic.

Reply:
yep already added to the other PR.. I've tested it out and works fine

Reply:
Sweet, would be great if you could open a PR to upgrade https://github.com/getumbrel/docker-tor as well. Ideally, we should only be using `getumbrel/*` images, which also reminds me that we should switch over to `getumbrel` forks of both `bitcoind` and `lnd` images of `lncm` too.

Reply:
Sure. I don't own that tor project though so can't make releases and stuff when stuff is upgraded